pypeit/core/wavecal/identify.py

```diff
@@ -0,0 +1,770 @@
+""" Module for finding patterns in arc line spectra
+"""
+import os
+import copy
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+from matplotlib.colors import LinearSegmentedColormap, Normalize
+from matplotlib.cm import ScalarMappable
+import matplotlib.transforms as mtransforms
+from matplotlib.widgets import Button, Slider
+
+matplotlib.use('Qt5Agg')
+
+from pypeit.par import pypeitpar
+from pypeit.core.wavecal import fitting, waveio, wvutils
+from pypeit import msgs
+
+
+class Identify(object):
+    """
+    Manually identify arc lines.
+
+    """
+
+    def __init__(self, canvas, ax, axr, axi, spec, specres, detns, line_lists, par, slit=0):
+        """ Controls for the Identify task in PypeIt. The main goal of this routine is to
+        manually identify arc lines to be used for wavelength calibration.
+
+        Parameters
```

Comment:
If you are willing to switch over to Google-style docs, please do.

Someone will do so eventually, if you don't.